### ISSUE TITLE
Fix copy records in PPXML writer

### ISSUE THREAD

**Issue by nichtich** [gbv/PICA-Data#71]
I tried to rewrite the PPXML writer and stumbled upon how to express multiple level 2 records belonging to the same level 1 record. here is a minimal example:

~~~
101@ $a42
203@/01 $0123
203@/02 $0456
~~~

Current serialization in PPXML is:

~~~xml
<ppxml:record xmlns:ppxml="http://www.oclcpica.org/xmlns/ppxml-1.0">
    <ppxml:global opacflag="" status=""/>
    <ppxml:owner iln="42">
      <ppxml:local>
        <ppxml:tag id="101@" occ="">
          <ppxml:subf id="a">42</ppxml:subf>
        </ppxml:tag>
      </ppxml:local>
      <ppxml:copy occ="2" opacflag="" status="" epn="123">
        <ppxml:tag id="203@" occ="1">
          <ppxml:subf id="0">123</ppxml:subf>
        </ppxml:tag>
        <ppxml:tag id="203@" occ="2">
          <ppxml:subf id="0">456</ppxml:subf>
        </ppxml:tag>
      </ppxml:copy>
    </ppxml:owner>
  </ppxml:record>
~~~

So two copy records with EPN and 123 and EPN 456 are combined under one `copy` element with same attributes `occ` and `epn`? I think this should better be:

~~~xml
<ppxml:record xmlns:ppxml="http://www.oclcpica.org/xmlns/ppxml-1.0">
    <ppxml:global opacflag="" status=""/>
    <ppxml:owner iln="42">
      <ppxml:local>
        <ppxml:tag id="101@" occ="">
          <ppxml:subf id="a">42</ppxml:subf>
        </ppxml:tag>
      </ppxml:local>
      <ppxml:copy occ="1" opacflag="" status="" epn="123">
        <ppxml:tag id="203@" occ="1">
          <ppxml:subf id="0">123</ppxml:subf>
        </ppxml:tag>
      </ppxml:copy>
      <ppxml:copy occ="2" opacflag="" status="" epn="456">
        <ppxml:tag id="203@" occ="2">
          <ppxml:subf id="0">456</ppxml:subf>
        </ppxml:tag>
      </ppxml:copy>
    </ppxml:owner>
  </ppxml:record>
~~~

**Comment by nichtich** [comment]
Done with https://github.com/gbv/PICA-Data/commit/a14ac1ebeda2229088dbcacf882aca6d1275ed0f at the `encoding` branch.

**Comment by jorol** [comment]
Sorry, I'm on vacation and only check my mails from time to time...

The current implementation from DNB creates an <owner>/<local>/<copy> for each copy of an ILN:

```xml
<ppxml:owner iln="11">
    <ppxml:local>
        <ppxml:tag id="101@" occ="">
            <ppxml:subf id="a">11</ppxml:subf>
        </ppxml:tag>
    </ppxml:local>
    <ppxml:copy epn="082832161" occ="1" opacflag="" status="">
        <ppxml:tag id="203@" occ="2">
            <ppxml:subf id="0">082832161</ppxml:subf>
        </ppxml:tag>
    </ppxml:copy>
</ppxml:owner>
<ppxml:owner iln="11">
    <ppxml:local>
        <ppxml:tag id="101@" occ="">
            <ppxml:subf id="a">11</ppxml:subf>
        </ppxml:tag>
    </ppxml:local>
    <ppxml:copy epn="082832188" occ="1" opacflag="" status="">
        <ppxml:tag id="203@" occ="3">
            <ppxml:subf id="0">082832188</ppxml:subf>
        </ppxml:tag>
    </ppxml:copy>
</ppxml:owner>
<ppxml:owner iln="11">
    <ppxml:local>
        <ppxml:tag id="101@" occ="">
            <ppxml:subf id="a">11</ppxml:subf>
        </ppxml:tag>
    </ppxml:local>
    <ppxml:copy epn="082832196" occ="1" opacflag="" status="">
        <ppxml:tag id="203@" occ="4">
            <ppxml:subf id="0">082832196</ppxml:subf>
    </ppxml:copy>
</ppxml:owner>
```

Example: http://services.dnb.de/sru/zdb?version=1.1&operation=searchRetrieve&query=num%3D3181-1&recordSchema=PicaPlus-xml

**Comment by nichtich** [comment]
Well, this might require changes in the PPXML reader to not repeat level 1 fields with same ILN. I need to check back with colleagues!

**Comment by nichtich** [comment]
Fixed writing PPXML in 1.12 but proper reading of records with multiple copy should be solved by implementing #5 including a general function to group records like:

~~~
101@ $a1
203@/01 1$a111 
101@ $a2
203@/01 $a222
101@ $a1
111X $yz
203@/02 $a333
~~~

into

~~~
101@ $a1
111X $yz
203@/01 1$a111 
203@/02 $a333
101@ $a2
203@/01 $a222
~~~
